### PR TITLE
Moving some metrics about API requests to operator-utils

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

Creating a `metrics` package that can be used from any of the Go backend services. It also includes the middleware for adding it to an HTTP server.

This PR also includes some useful tooling in order to test the API metrics present on this repository without depending on other HTTP server implementations like Aggregator's one

## Type of change
- New feature (non-breaking change which adds functionality)

## Testing steps
Regular CI

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+*/
+
+// Package metrics contains all metrics that needs to be exposed to Prometheus
+// and indirectly to Grafana. Currently, the following metrics are exposed:
+//
+// api_endpoints_requests - number of requests made for each REST API endpoint
+//
+// api_endpoints_response_time - response times for all REST API endpoints
+//
+// consumed_messages - total number of messages consumed from selected broker
```
Comment (Sergey1011010) [inline metrics/metrics.go:24]: This comment is lil bit outdated

Comment (joselsegura) [reply metrics/metrics.go]: Oh, nice catch. Updating

File: metrics/server_config.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func (writer loggingResponseWriter) WriteHeader(statusCode int) {
+	writer.ResponseWriter.WriteHeader(statusCode)
+	APIResponseStatusCodes.With(
+		prometheus.Labels{"status_code": fmt.Sprint(statusCode)},
+	).Inc()
+}
+
+func logRequestHandler(writer http.ResponseWriter, request *http.Request, nextHandler http.Handler) {
```
Comment (Sergey1011010) [inline metrics/server_config.go:38]: why is it in metrics package? It's more like something related to a server.

Comment (joselsegura) [reply metrics/server_config.go:38]: In fact it is related to both. It is a middleware for an HTTP server, but it simply do the "Inc" for the given metrics.

I think it is a good place, but probably bad file name... Let's see what is the opinion of the rest of the people :-)

File: go.sum [hunk 3]
```diff
@@ -1,58 +1,143 @@
 github.com/PuerkitoBio/goquery v1.5.1/go.mod h1:GsLWisAFVj4WgDibEWF4pvYnkVQBpKBKeU+7zCJoLcc=
+github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc/go.mod h1:LOuyumcjzFXgccqObfd/Ljyb9UuFJ6TxHnclSeseNhc=
+github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751/go.mod h1:LOuyumcjzFXgccqObfd/Ljyb9UuFJ6TxHnclSeseNhc=
+github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf/go.mod h1:ybxpYRFXyAe+OPACYpWeL0wqObRcbAqCMya13uyzqw0=
+github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4/go.mod h1:ybxpYRFXyAe+OPACYpWeL0wqObRcbAqCMya13uyzqw0=
 github.com/andybalholm/cascadia v1.1.0/go.mod h1:GsXiBklL0woXo1j/WYWtSYYC4ouU9PqHO0sqidkEA4Y=
+github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973/go.mod h1:Dwedo/Wpr24TaqPxmxbtue+5NUziq4I4S80YR8gNf3Q=
+github.com/beorn7/perks v1.0.0/go.mod h1:KWe93zE9D1o94FZ5RNwFwVgaQK1VOXiVxmqh+CedLV8=
+github.com/beorn7/perks v1.0.1 h1:VlbKKnNfV8bJzeqoa4cOKqO6bYr3WgKZxO8Z16+hsOM=
+github.com/beorn7/perks v1.0.1/go.mod h1:G2ZrVWU2WbWT9wwq4/hrbKbnv/1ERSJQ0ibhJ6rlkpw=
```
Comment (tisnik) [inline go.sum:10]: these dependencies are a bit strange -> Golang ecosystem is becoming NPM one :)

File: metrics/server_config.go (hunk 2 above)
Comment (Sergey1011010) [reply metrics/server_config.go:38]: I'd move it to httputils package, but it's up to you :)

Comment (joselsegura) [reply metrics/server_config.go]: Nice idea. Should we rename the `http` directory to `httputils`?

**Pull request merged**